### PR TITLE
dash/hc-multiple-connectors

### PR DESCRIPTION
**Pull request by vazonik**
Added ability to set multiple connectors for the Highcharts Component.

---

TO DO:
- [x] Implement multiple connectors logic.
- [x] Customize existing Highcharts Component synchronizations to work for multiple connectors.
- [x] Update existing demos if needed (expecially custom components ones).
- [x] Add multiple connectors demo.
- [x] Update cypress tests.
- [x] Update karma tests.
- [x] Check whether the update of the column assignment option in the editmode sidebar works.
- [x] Update docs.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206840051983379

### CODE REVIEW

**Comment by sebastianbochan** [comment]
One suggestion here: Connectors are configurable by the sidebar, so we should replace the current `selector` with multiple version.

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Tremendous job! I'm glad you've devided the syncs, thanks 😉

File: docs/dashboards/highcharts-component.md [hunk 1]
```diff
@@ -177,6 +177,27 @@ columnAssignment: [{
 ```
 <iframe style="width: 100%; height: 600px; border: none;" src=https://www.highcharts.com/samples/embed/dashboards/components/highcharts-column-assignment-keys-data allow="fullscreen"></iframe>
 
+### Multiple connectors
+
+Highcharts Component supports also more than one data source. This can be achieved by setting the `connector` option to an array of objects rather than a single object.
```
Comment (jomar-honsi) [inline docs/dashboards/highcharts-component.md:182]: Suggested rewrite: The Highcharts component also supports ...

Comment (sebastianbochan) [reply docs/dashboards/highcharts-component.md:182]: I would also rephrase the second sentence. I suggest:
 `That means the connector option should be configured as an array of objects...`

File: ts/Dashboards/Components/KPIComponent/KPISyncs/KPIExtremesSync.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+        };
+
+        const registerCursorListeners = (): void => {
+            const { dataCursor: cursor } = board;
+
+            if (!cursor) {
+                return;
+            }
+            const table = this.connectorHandlers?.[0]?.connector?.table;
```
Comment (sebastianbochan) [inline ts/Dashboards/Components/KPIComponent/KPISyncs/KPIExtremesSync.ts:76]: In some places you use `connectorHandlers?.[0]?` in other you use different construction (or use extra condition) like above.

Can we unify it? 

ps. I'm not convinced to the notation: `connectorHandlers?.[0]?` too many `optionals`, so I prefer to add a condition then. Up to you.

File: ts/Dashboards/Components/DataGridComponent/DataGridComponent.ts [hunk 3]
```diff
@@ -160,8 +151,11 @@ class DataGridComponent extends Component {
         this.innerResizeTimeouts = [];
 
 
-        this.on('afterSetConnector', (e: any): void => {
-            this.disableEditingModifiedColumns(e.connector);
+        this.on('afterSetConnectors', (e: any): void => {
+            const connector = e.connectorHandlers?.[0]?.connector;
```
Comment (sebastianbochan) [inline ts/Dashboards/Components/DataGridComponent/DataGridComponent.ts:155]: The `connectorHandler` is always an array (default definition in the `Component.ts`), so what is the use case of missing the array type?

**Review by jomar-honsi — APPROVED** [review]

File: docs/dashboards/highcharts-component.md [hunk 4]
```diff
@@ -177,6 +177,27 @@ columnAssignment: [{
...
+
+Code sample:
+```js
+components: [{
+    type: 'Highcharts',
+    connector: [{
+        id: 'connector-1',
```
[the remark this answers is not in the document]
Comment (Vazonik) [reply docs/dashboards/highcharts-component.md:189]: I'm not sure what you mean by `seriesId` in this case. 🤔

**Review by jomar-honsi — APPROVED** [review]

**Review by jomar-honsi — APPROVED** [review]

File: docs/dashboards/highcharts-component.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (karolkolodziej) [reply docs/dashboards/highcharts-component.md:189]: My idea was how to connect a given series with the connector.

**Review by karolkolodziej — APPROVED** [review]

**Review by sebastianbochan — APPROVED** [review]

**Review by stitot — APPROVED** [review]
Have messed around with this, and can't find any biggies. Great work!

**Pull request merged**